PowerEditor/src/NppBigSwitch.cpp

```diff
@@ -1959,20 +1959,37 @@ LRESULT Notepad_plus::process(HWND hwnd, UINT message, WPARAM wParam, LPARAM lPa
 			}
 			else
 			{
-				if ((HWND(wParam) == _mainEditView.getHSelf()) || (HWND(wParam) == _subEditView.getHSelf()))
+				HWND activeViewHwnd = reinterpret_cast<HWND>(wParam);
+
+				if ((activeViewHwnd == _mainEditView.getHSelf()) || (activeViewHwnd == _subEditView.getHSelf()))
 				{
-					if ((HWND(wParam) == _mainEditView.getHSelf()))
+					if (activeViewHwnd == _mainEditView.getHSelf())
 						switchEditViewTo(MAIN_VIEW);
 					else
 						switchEditViewTo(SUB_VIEW);
 
-					POINT p;
-					::GetCursorPos(&p);
 					ContextMenu scintillaContextmenu;
+					
 					std::vector<MenuItemUnit>& tmp = nppParam.getContextMenuItems();
 					bool copyLink = (_pEditView->getSelectedTextCount() == 0) && _pEditView->getIndicatorRange(URL_INDIC);
 					scintillaContextmenu.create(hwnd, tmp, _mainMenuHandle, copyLink);
+					
+					POINT p;
+					if (lParam == -1)

```

Comment:
I cannot find the Microsoft's document for indicating the meaning of lParam value "-1" for  WM_CONTEXTMENU.
OTOH, ***If the context menu is generated from the keyboard for example, if the user types SHIFT+F10 then the x- and y-coordinates are -1 and the application should display the context menu at the location of the current selection rather than at (xPos, yPos).***

https://learn.microsoft.com/en-us/windows/win32/menurc/wm-contextmenu

Reply:
It is in the _Remarks_ section:  https://learn.microsoft.com/en-us/windows/win32/menurc/wm-contextmenu#:~:text=the%20VK_APPS%20key.-,If%20the%20context%20menu,-is%20generated%20from

Reply:
Ready for review again.